blivet/devices/lvm.py

```diff
@@ -1485,19 +1512,30 @@ def lvs(self):
         """ A list of this pool's LVs """
         return self._lvs[:]     # we don't want folks changing our list
 
-    @property
-    def vg_space_used(self):
-        # TODO: what about cached thin pools?
+    @util.requires_property("is_thin_pool")
+    def autoset_md_size(self):
+        """ If self._metadata_size not set already, it calculates the recommended value
+        and sets it while subtracting the size from self.size.
+
+        """
 
-        space = self.data_vg_space_used + self.metadata_vg_space_used
+        if self._metadata_size != 0:
+            return  # Metadata size already set
 
-        # We need to make the nonexisting thin pools pretend to be bigger so
-        # that their padding is not eaten by some other LV, but we need to
-        # provide consistent information with the LVM tools for existing
-        # thin pools.
-        if not self.exists:
-            space += Size(blockdev.lvm.get_thpool_padding(space, self.vg.pe_size))
-        return space
+        log.debug("Auto-setting thin pool metadata size")
+
+        # we need to know chunk size to calculate recommended metadata size
+        if self._chunk_size == 0:
+            self._chunk_size = Size(blockdev.LVM_DEFAULT_CHUNK_SIZE)
+            log.debug("Using default chunk size: %s", self._chunk_size)
+
+        self._metadata_size = Size(blockdev.lvm.get_thpool_meta_size(self._size,
+                                                                     self._chunk_size,
+                                                                     100))  # snapshots
+        log.debug("Recommended metadata size: %s", self._metadata_size)
+
+        log.debug("Adjusting size from %s to %s", self.size, self.size - self._metadata_size)
+        self.size = self.size - self._metadata_size
```

Comment:
The idea here being that taking metadata space from the requested lv has less potential to cause a difficult situation since users might, for example, expect to be able to allocate lvs whose combined size equals the total vg size, right?